src/summary.cpp

```diff
@@ -0,0 +1,124 @@
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <numeric>
+#include <ratio>
+
+#include <boost/filesystem.hpp>
+
+#include <lo2s/config.hpp>
+#include <lo2s/summary.hpp>
+#include <lo2s/util.hpp>
+
+extern "C" {
+#include <sys/time.h>
+}
+
+namespace lo2s
+{
+
+Summary& summary()
+{
+    static Summary s;
+    return s;
+}
+
+Summary::Summary()
+: start_wall_time_(std::chrono::steady_clock::now()), num_wakeups_(0),
+  thread_count_(0), trace_dir_(""), exit_code_(0)
+{
+}
+
+void Summary::register_process(pid_t pid)
+{
+    pids_.emplace(pid);
+}
+
+void Summary::record_perf_wakeups(std::size_t num_wakeups)
+{
+    num_wakeups_ += num_wakeups;
+}
+
+void Summary::set_exit_code(int exit_code)
+{
+    exit_code_ = exit_code;
+}
+
+void Summary::add_thread()
+{
+    thread_count_++;
+}
+
+std::string pretty_print_bytes(int trace_size)
```

Comment:
please use std::size_t trace_size. int has only 32-bit, thus would only allow for 2GB.